### PR TITLE
:warning: Remove TMC/Syncer

### PR DESCRIPTION
**Pull request by xrstf**
## Summary
This supersedes #2955, which would have to be rebased and was also removing such a small portion that it's effectively meaningless.

Part of #2954.

### CODE REVIEW

File: pkg/admission/pathannotation/pathannotation_admission_test.go [hunk 1]
```diff
@@ -125,16 +124,6 @@ func TestPathAnnotationAdmit(t *testing.T) {
 			validateAdmissionObject: objectHasPathAnnotation("root:foo"),
 		},
 
-		{
-			name:                    "happy path: a Location is annotated with a path",
-			admissionVerb:           admission.Create,
-			admissionResource:       schedulingv1alpha1.SchemeGroupVersion.WithResource("locations"),
-			admissionObject:         &schedulingv1alpha1.Location{},
```
Comment (sttts) [inline pkg/admission/pathannotation/pathannotation_admission_test.go:132]: can we just switch to another kind?

File: pkg/syncer/shared/helpers.go [hunk 2]
```diff
@@ -29,26 +29,11 @@ import (
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/sets"
-
-	workloadv1alpha1 "github.com/kcp-dev/kcp/sdk/apis/workload/v1alpha1"
 )
 
 // SyncableClusterScopedResources holds a set of cluster-wide GVR that are allowed to be synced.
 var SyncableClusterScopedResources = sets.New[string](schema.GroupVersionResource{Group: "", Version: "v1", Resource: "persistentvolumes"}.String())
```
Comment (sttts) [inline pkg/syncer/shared/helpers.go:35]: this surprises that not everything is removed from this package.

File: test/e2e/framework/config.go [hunk 3]
```diff
@@ -100,10 +95,9 @@ func registerFlags(c *testConfig) {
 	flag.StringVar(&c.kcpKubeconfig, "kcp-kubeconfig", "", "Path to the kubeconfig for a kcp server.")
 	flag.Var(cliflag.NewMapStringString(&c.shardKubeconfigs), "shard-kubeconfigs", "Paths to the kubeconfigs for a kcp shard server in the format <shard-name>=<kubeconfig-path>. If unset, kcp-kubeconfig is used.")
 	flag.StringVar(&c.pclusterKubeconfig, "pcluster-kubeconfig", "", "Path to the kubeconfig for a kubernetes cluster to sync to. Requires --syncer-image.")
```
Comment (sttts) [inline test/e2e/framework/config.go:97]: this should go too

File: pkg/admission/pathannotation/pathannotation_admission_test.go (hunk 1 above)
Comment (xrstf) [reply pkg/admission/pathannotation/pathannotation_admission_test.go:132]: We can. I restored the testcase.

File: tmc/pkg/server/options/options.go [hunk 4]
```diff
@@ -49,9 +46,8 @@ type CompletedOptions struct {
 // NewOptions creates a new Options with default parameters.
 func NewOptions(rootDir string) *Options {
 	o := &Options{
-		Core:                 *kcpcoreoptions.NewOptions(rootDir),
-		TmcControllers:       *NewTmcControllers(),
-		TmcVirtualWorkspaces: *tmcvirtualoptions.NewOptions(),
+		Core:           *kcpcoreoptions.NewOptions(rootDir),
+		TmcControllers: *NewTmcControllers(),
```
[the remark this answers is not in the document]
Comment (xrstf) [reply tmc/pkg/server/options/options.go]: Nothing anymore, so I now removed the tmc stuff entirely and rewrote the kcp cmds to use pkg/server directly.

**Comment by xrstf** [comment]
https://github.com/openshift/release/pull/39918 should remove the parts of the CI configuration that make this PR break right now. Until that is merged, I guess this PR is blocked unless we want to add dummy contents back into this repo to make the jobs pretend to work.

**Pull request merged**